code/modules/projectiles/gun_system.dm

```diff
@@ -176,6 +176,8 @@
 	var/detach_delay = 0 SECONDS
 	///Icon state used for an added overlay for a sentry. Currently only used in Build-A-Sentry.
 	var/placed_overlay_iconstate = "rifle"
+	///The beam icon_state for hitscan gun
+	var/hitscan_gun_effect_icon = "beam"
```

Comment:
these should be on the ammo since you know its the ammo type thats important for the icon